ocf_data_sampler/numpy_batch/datetime_features.py

```diff
@@ -0,0 +1,42 @@
+"""Datapipes to trigonometric date and time to NumpyBatch"""
+
+import numpy as np
+import pandas as pd
+from numpy.typing import NDArray
+
+
+def _get_date_time_in_pi(
+    dt: pd.DatetimeIndex,
+) -> tuple[NDArray[np.float64], NDArray[np.float64]]:
+    """
+    Change the datetimes, into time and date scaled in radians
+    """
+
+    day_of_year = dt.dayofyear
+    minute_of_day = dt.minute + dt.hour * 60
+
+    # converting into positions on sin-cos circle
+    time_in_pi = (2 * np.pi) * (minute_of_day / (24 * 60))
+    date_in_pi = (2 * np.pi) * (day_of_year / 365)
+
+    return date_in_pi, time_in_pi
+
+
+def make_datetime_numpy_batch(datetimes: pd.DatetimeIndex, key_prefix: str = "wind") -> dict:
```

Comment:
Potentially update main function to include input validation - also to simplify the dict construction via literal. Feel this improves maintenance ease a little overall. As example:

```python
def make_datetime_numpy_batch(datetimes: pd.DatetimeIndex, key_prefix: str = "wind") -> dict:
    """
    Make a dict of datetime features using sin/cos transformations
    """
    if datetimes.empty:
        raise ValueError("Input datetimes is empty")

    # Obtain scaled radians for date and time
    date_in_pi, time_in_pi = _get_date_time_in_pi(datetimes)

    # Create dict
    return {
        f"{key_prefix}_date_sin": np.sin(date_in_pi),
        f"{key_prefix}_date_cos": np.cos(date_in_pi),
        f"{key_prefix}_time_sin": np.sin(time_in_pi),
        f"{key_prefix}_time_cos": np.cos(time_in_pi),
    }
```